src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -814,9 +979,110 @@ export class ShadowGenerator implements IShadowGenerator {
         this._applyFilterValues();
     }
 
+    private _initCascades(): void {
+        let camera = this._scene.activeCamera;
+        if (!camera) {
+            return;
+        }
+
+        // inits and sets all static params related to CSM
+        let engine = this._scene.getEngine();
+        this._viewSpaceFrustums = [];
+        this._viewSpaceBoundingSpheres = [];
+        let nearx = 0;
+        let neary = 0;
+        let farx = 0;
+        let fary = 0;
+
+        // get all internal camera cascaded frustum points
+        let breaks = this._frustumSplit(this.cascades, camera.minZ, this._csmFrustumLength, 0.5);

```

Comment:
0.5 should be the default value for lambda but kept configurable as a property

Reply:
ok, fixed to ```private _csmLambda = 0.5;```.

Reply:
Good but would still need a public property to be able to modify it ?

This would dynamically recreate the matrices and so on.

Reply:
ok, fixed by adding in public getters and setters.